packages/sveltekit/src/client/handleError.ts

```diff
@@ -11,23 +11,39 @@ function defaultErrorHandler({ error }: Parameters<HandleClientError>[0]): Retur
   });
 }
 
-// TODO: add backwards-compatible type for kit 1.x (soon)
 type HandleClientErrorInput = Parameters<HandleClientError>[0];
 
+/**
+ * Backwards-compatible HandleServerError Input type for SvelteKit 1.x and 2.x
+ * `message` and `status` were added in 2.x.
+ * For backwards-compatibility, we make them optional
+ *
+ * @see https://kit.svelte.dev/docs/migrating-to-sveltekit-2#improved-error-handling
+ */
+type SafeHandleServerErrorInput = Omit<HandleClientErrorInput, 'status' | 'message'> &
+  Partial<Pick<HandleClientErrorInput, 'status' | 'message'>>;
+
 /**
  * Wrapper for the SvelteKit error handler that sends the error to Sentry.
  *
  * @param handleError The original SvelteKit error handler.
  */
 export function handleErrorWithSentry(handleError: HandleClientError = defaultErrorHandler): HandleClientError {
-  return (input: HandleClientErrorInput): ReturnType<HandleClientError> => {
-    captureException(input.error, {
-      mechanism: {
-        type: 'sveltekit',
-        handled: false,
-      },
-    });
+  return (input: SafeHandleServerErrorInput): ReturnType<HandleClientError> => {
+    const { status, message } = input;
+    const isNotFoundError = status === 404 && message === 'Not Found';
```

Comment:
l: might make sense to extract this into function to use between client/server in case this logic needs to change in the future.

Reply:
Would have agreed but now that I simplified the check here and in #9901, I think it's no longer worth extract checking for a 404 status.